src/types.rs

```diff
@@ -104,6 +104,28 @@ impl StructType {
             ty.into()
         })
     }
+    /// Make a new named struct without defining its fields yet.
+    ///
+    /// You can use this opaque struct without defining its fields (for creating recursive types, etc.). When you want to define its fields (which may include itself), use set_elements().
+    pub fn new_opaque<'a>(context: &'a Context, name: &str) -> &'a StructType {
+        util::with_cstr(name, |name| unsafe {
+            core::LLVMStructCreateNamed(context.into(), name).into()
+        })
+    }
+    /// Set the elements that make up this struct.
+    ///
+    /// Can only be called once, and only on a StructType created through the new_opaque() function.
+    /// Returns Err(()) if the struct is not opaque.
+    pub fn set_elements<'a>(&self, fields: &[&'a Type], packed: bool) -> Result<(), ()> {
```

Comment:
I'm not sure about returning a `Result<(), ()>`, but I'm not sure what the rust convention is when an operation has no return value regardless of whether it succeeds or fails.

And returning a `bool` seems wrong.